### PR TITLE
Save stack arguments size in InlinedCallFrame.m_Datum on x86 to handle callee's popping of arguments

### PR DESCRIPTION
**Pull request by fadimounir**
Fix for https://github.com/dotnet/runtime/issues/2230.

### CODE REVIEW

**Comment by fadimounir** [comment]
cc @dotnet/jit-contrib

**Comment by fadimounir** [comment]
I've been able to repro the issue with some stress testing before the fix. I currently have a couple of test instances running in a loop, and so far there are no crashes/asserts

File: src/coreclr/src/jit/lower.cpp [hunk 1]
```diff
@@ -3688,9 +3688,18 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
         GenTree* frameAddr =
             new (comp, GT_LCL_VAR_ADDR) GenTreeLclVar(GT_LCL_VAR_ADDR, TYP_BYREF, comp->lvaInlinedPInvokeFrameVar);
 
+#ifdef _TARGET_X86_
+        // On x86 targets, indirect calls need the size of the stack args in InlinedCallFrame.m_Datum.
```
Comment (jkotas) [inline src/coreclr/src/jit/lower.cpp]: indirect calls -> PInvoke calls. This applies to both direct and indirect calls.

Comment (fadimounir) [reply src/coreclr/src/jit/lower.cpp]: Yes. That's a copy-paste bug in the comment text. Thanks for pointing it out

**Review by jkotas — APPROVED** [review]
Thanks!

File: src/coreclr/src/jit/lower.cpp [hunk 2]
```diff
@@ -3688,9 +3688,18 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
         GenTree* frameAddr =
             new (comp, GT_LCL_VAR_ADDR) GenTreeLclVar(GT_LCL_VAR_ADDR, TYP_BYREF, comp->lvaInlinedPInvokeFrameVar);
 
+#ifdef _TARGET_X86_
```
Comment (jkotas) [inline src/coreclr/src/jit/lower.cpp]:
```suggestion
#ifdef TARGET_X86
```
The ifdefs were renamed recently.

Comment (jkotas) [reply src/coreclr/src/jit/lower.cpp]: > This is for Windows only

For clarity, you can also move all logic inside `InlinedCallFrame::UpdateRegDisplay` that deals with stack argument sizes under `UNIX_X86_ABI`.

Comment (fadimounir) [reply src/coreclr/src/jit/lower.cpp]: Ok, so what's the Unix ABI? Does the caller pop the arguments from the stack?

Comment (jkotas) [reply src/coreclr/src/jit/lower.cpp]: Right

**Review by jkotas — APPROVED** [review]
Thanks!

**Comment by jkotas** [comment]
src/coreclr/src/jit/lower.cpp has conflicts...

**Pull request merged**